File: v2_single_responsibility/python/macbook_v2.py
class LaptopBasicProperty: 
    cpu = ""
    size = 0

class LaptopExtensionProperty:
    resolution = ""
    
class LaptopBasicFunction:
    def startUp(self): pass
    def shutDown(self): pass

class LaptopExtensionFunction:
    def sleep(self): pass

class MacLaptopBasicProperty(LaptopBasicProperty):
    cpu = ""
    size = 0

class MacLaptopExtensionProperty(LaptopExtensionProperty):
    resolution = ""
    
class MacLaptopBasicFunction(LaptopBasicFunction):
    def startUp(self): pass
    def shutDown(self): pass

class MacLaptopExtensionFunction(LaptopExtensionFunction):
    def sleep(self): pass

class MacBookV2:
    basic_property: MacLaptopBasicProperty
    basic_function: MacLaptopBasicFunction
    extension_property: MacLaptopExtensionProperty
    extension_function: MacLaptopExtensionFunction
    
    def __init__(self, bp: MacLaptopBasicProperty, bf: MacLaptopBasicFunction, ep: MacLaptopBasicFunction, ef: MacLaptopExtensionFunction):
        self.basic_property = bp
        self.basic_function = bf
        self.extension_property = ep
        self.extension_function = ef

MacBookV2(MacLaptopBasicProperty(), MacLaptopBasicFunction(), MacLaptopBasicFunction(), MacLaptopExtensionFunction())

File: v2_single_responsibility/java/MacBookV2.java
public class MacBookV2 {
	private MacbookBasicProperty basicProperty;
	private MacbookExtensionProperty extensionProperty;
	private MacbookBasicFunction basicFunction;
	private MacbookExtensionFunction extensionFunction;

    public static void main(String[] args) {
        System.out.println("Hello");
    }

	public MacBookV2(MacbookBasicProperty bp, MacbookExtensionProperty ep,  MacbookBasicFunction bf, MacbookExtensionFunction ef) {
		this.basicProperty = bp;
		this.extensionProperty = ep;
		this.basicFunction = bf;
        this.extensionFunction = ef;
	}
}


class MacbookBasicProperty implements LaptopBasicProperty {
	String cpu = "";
	int size = 0;
}
class MacbookExtensionProperty implements LaptopExtensionProperty {
	String resolution = "";
}
class MacbookBasicFunction implements LaptopBaseFunction {
	public void startUp() {};
	public void shutDown() {};
}
class MacbookExtensionFunction implements LaptopExtentionFunction {
	public void sleep() {};
}



interface LaptopBasicProperty {
	String cpu = "";
	int size = 0;
}
interface LaptopExtensionProperty {
	String resolution = "";
}
interface LaptopBaseFunction {
	void startUp();
	void shutDown();
}
interface LaptopExtentionFunction {
	void sleep();
}
